crates/miden-lib/src/transaction/script_builder/test.rs

```diff
@@ -0,0 +1,188 @@
+use miden_objects::{
+    account::{AccountBuilder, AccountType},
+    asset::{Asset, FungibleAsset, TokenSymbol},
+    crypto::{
+        dsa::rpo_falcon512::PublicKey,
+        rand::{FeltRng, RpoRandomCoin},
+    },
+    note::{
+        NoteAssets, NoteExecutionHint, NoteExecutionMode, NoteInputs, NoteMetadata, NoteRecipient,
+        NoteScript, NoteTag, NoteType, PartialNote,
+    },
+    transaction::TransactionScript,
+    utils::word_to_felts_string,
+    Digest, Felt, ONE, ZERO,
+};
+
+use crate::{
+    account::{
+        auth::RpoFalcon512, faucets::BasicFungibleFaucet, interface::AccountInterface,
+        wallets::BasicWallet,
+    },
+    transaction::{TransactionKernel, TransactionScriptBuilder},
+};
+
+/// Tests the correctness of the send_note script generation in case the sending account has the
+/// [BasicWallet] interface.
+#[test]
+fn test_send_note_script_basic_wallet() {
+    let mock_public_key = PublicKey::new([ZERO, ONE, Felt::new(2), Felt::new(3)]);
```

Comment:
I think the tests for these scripts would be more useful if the scripts would get executed. Right now, the scripts could have subtle problems like leaving too many elements on the stack and we wouldn't notice. Executing the scripts would also make the test more maintenance friendly. If we change something about the script in the builder right now, we would have to make the equivalent change in this test here. If the test was executing the script and we changed the script but the behavior stays the same, then no changes would be needed.